include/data_buffer.h

```diff
@@ -0,0 +1,70 @@
+#pragma once
+#include <string>
+#include <vector>
+
+namespace reinforcement_learning { namespace utility {
+
+/*
+     * Data buffer used for serialized messages.  Data buffer
+     * consists of 2 parts.  1) preamble 2) body
+     *
+     * body: can start somewhere in the middle and end before the
+     * buffer ends.
+     * preamble: is n bytes immediately before the body.
+     * preamble is usually 8 bytes and sometimes zero if there is no preamble
+     */
+class data_buffer {
+public:
+  using value_type = unsigned char;
+
+  explicit data_buffer(size_t body_size = 1024);
+
+  // Get a pointer to beginning of preamble
+  value_type *preamble_begin();
+
+  constexpr size_t preamble_size() const;
```

Comment:
👍

Reply:
Had to get rid of constexpr here.  gcc complained that the class does not have a trivial destructor.